pip/commands/list.py

```diff
@@ -110,6 +130,22 @@ def run(self, options, args):
                 "no longer has any effect.",
                 RemovedInPip10Warning,
             )
+
+        # will this work in all usage cases?
+        if '--columns' in sys.argv and '--no-columns' in sys.argv:
+            raise CommandError(
+                "Options --columns and --no-columns cannot be combined.")
+
```

Comment:
I don't think we need this check. Consider a user with a shell alias ```pipls=pip list --columns```. That user may legitimately want to do ```pipls --no-columns```.

Also, it's a corner case and I just don't think the check is worth the effort.